### PR TITLE
Added check for demo app patching to CI.

### PR DESCRIPTION
**Pull request by Martinsos**
Resolves https://github.com/wasp-lang/open-saas/issues/201

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -26,11 +26,19 @@ jobs:
         with:
           node-version: '20'
 
+      - name: Docker setup
```
Comment (Martinsos) [inline .github/workflows/e2e-tests.yml:29]: I moved this and some other steps to make it more like "first setup", then "check patching", then "run e2e tests".

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -43,44 +51,41 @@ jobs:
...
           cp .env.server.example .env.server && cp .env.client.example .env.client
 
-      - name: Cache global node modules
-        uses: actions/cache@v4
-        with:
-          path: ~/.npm
-          key: node-modules-${{ runner.os }}-${{ hashFiles('template/app/package-lock.json') }}-${{ hashFiles('template/e2e-tests/package-lock.json') }}-wasp${{ env.WASP_VERSION }}-node${{ steps.setup-node.outputs.node-version }}
-          restore-keys: |
-            node-modules-${{ runner.os }}-
+      - name: Check that patching demo app (opensaas-sh) works
```
Comment (Martinsos) [inline .github/workflows/e2e-tests.yml:54]: This is the step I added!

File: opensaas-sh/tools/dope.sh [hunk 3]
```diff
@@ -123,6 +126,14 @@ recreate_derived_dir() {
     fi
 
     echo "DONE: generated ${DERIVED_DIR}/"
+
+    if [ ${num_patches_failed} -gt 0 ]; then
+        echo -e "${RED_COLOR}${num_patches_failed} patches failed, look into generated files for merge conflicts.${RESET_COLOR}"
+        exit 1;
```
Comment (Martinsos) [inline opensaas-sh/tools/dope.sh:132]: I added this so that we return 1 as exit status/code if any of the patches fails, so that CI also fails then.

**Review by vincanger — APPROVED** [review]
ok looks good. tested it out locally. I'm going to merge it now and fix the diff :)

**Pull request merged**